### PR TITLE
Dl 718 update terraform aws data lake syntax

### PR DESCRIPTION
**Pull request by shayan-golafshani**
Updating syntax to Terraform 0.12

### CODE REVIEW

File: modules/emr/variables.tf [hunk 1]
```diff
@@ -1,100 +1,100 @@
...
 variable "task_instance_max_count" {
   description = "Max number of Task Nodes used on autoscale"
-  type        = "string"
+  type        = string
   default     = "4"
 }
 
 locals {
-  tags = "${merge(map("vendor", "segment"), var.tags)}"
+  tags = merge(tomap({"vendor"= "segment"}), var.tags)
```
Comment (hrdamani) [inline modules/emr/variables.tf:99]: nitpick. add a space after vendor.

File: modules/iam/variables.tf [hunk 2]
```diff
@@ -17,20 +17,20 @@ variable "segment_aws_accounts" {
...
 variable "tags" {
   description = "A map of tags to add to all resources. A vendor=segment tag will be added automatically."
-  type        = "map"
+  type        = map(string)
   default     = {}
 }
 
 locals {
-  tags = "${merge(map("vendor", "segment"), var.tags)}"
+  tags = merge(tomap({"vendor"= "segment"}), var.tags)
```
Comment (hrdamani) [inline modules/iam/variables.tf:35]: space after vendor

File: README.md [hunk 3]
```diff
@@ -122,8 +122,8 @@ module "emr" {
 #   for_each = local.glue_db_list
 #   name = each.key
 #   iam_roles = {
-#                datalake_role = "${module.iam.segment_datalake_iam_role_arn}", 
-#                emr_instance_profile_role = "${module.iam.iam_emr_instance_profile}"
+#                datalake_role = module.iam.segment_datalake_iam_role_arn,
```
Comment (hrdamani) [inline README.md:125]: can you add tabs befor = to make it fomatted

File: modules/emr/variables.tf (hunk 1 above)
Comment (shayan-golafshani) [reply modules/emr/variables.tf]: Fixed, thanks for the catch

File: modules/iam/variables.tf (hunk 2 above)
Comment (shayan-golafshani) [reply modules/iam/variables.tf]: Fixed, thanks hitisha

File: README.md (hunk 3 above)
Comment (shayan-golafshani) [reply README.md:125]: Yeah, taking a look now!

**Review by hrdamani — APPROVED** [review]

**Comment by shayan-golafshani** [comment]
Checked and validated that Datalake setup still works with reference pointing to my branch.

**Review by maggieyu-segment — APPROVED** [review]

**Pull request merged**